### PR TITLE
APIv4 - Deprecate option_value joins and display notices in Explorer

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Deprecates old way of joining on option values in favor of pseudoconstant lookups in APIv4. Displays those deprecation notices in the API Explorer.

Before
----------------------------------------
No deprecation notices.

After
----------------------------------------
Deprecation notices triggered and displayed in API Explorer.

### CODE REVIEW

**Comment by colemanw** [comment]
@eileenmcnaughton this is for the search builder project

File: Civi.php [hunk 1]
```diff
@@ -75,7 +75,7 @@ public static function lockManager() {
   }
 
   /**
-   * @return \Psr\Log\LoggerInterface
+   * @return \CRM_Core_Error_Log
```
Comment (eileenmcnaughton) [inline Civi.php:78]: I feel like this *could*not  be something else in future so Logger might be better

**Comment by eileenmcnaughton** [comment]
OK - yes - this completes our other change

File: Civi.php (hunk 1 above)
Comment (colemanw) [reply Civi.php:78]: Well currently it always returns CRM_Core_Error_Log but if that changes we can update the annotation.

**Pull request merged**